### DISCUSSION TITLE
number of communities at each level

### DISCUSSION THREAD

**Discussion by yingzwang** [microsoft/graphrag#1114] [Q&A]
After running GraphRAG indexing pipeline (with all default settings), the output communities have following stats,
|   level |   num_communities |   total_members |
|---------|-------------------|-----------------|
|       0 |                41 |            3286 |
|       1 |               253 |            4152 |
|       2 |               192 |            2600 |
|       3 |                 4 |             154 |

The Leiden algorithm produces 4 hierarchical levels. I count the number of communities at each level as "num_communities", and sum the number of entities over all communities at each level as "total_members". The result does not make sense to me. 

1) Why does level 0 have only 41 communities? I expect that the lower level the more communities.

2) Why are the total number of nodes different at each level? I expect that the total number of nodes should stay the same. They are just partitioned differntly at each level.

Did I misunderstand anything?

**Comment by natoverse** [comment]
1. The hierarchy definition is upside-down from how you're thinking about it. 0 is the root level of the tree, with the fewest, most coarse-grained communities. As the level increases, the communities subdivide and tend to create more. As you'll notice from your table, it isn't actually guaranteed that community count will increase with level, it's just a typical observation (see next bullet).
2. As the community level increases, they become tighter and more exclusive, so nodes begin to "orphan out". By the time we get to the finest level of communities, we frequently see a large number of nodes that are no longer close enough to a critical mass of nodes for any one community. It is also possible that there are not enough nodes that meet the connectedness criteria to form a cohesive community, so the quantity communities themselves can in fact begin to drop.

**Comment by yingzwang** [comment]
Thanks for the clarification. Now I understand that level 0 is most coarse, and when level increases it gets more fine-granular. But I still find my output at level 3 a bit weird. The community count increases from 0 to 1 as expected, and decreases a bit from 1 to 2 due to 'orphan' nodes. But at level 3 there're only 4 communities at the finest level. Is it reasonable that suddenly so many nodes get 'orphaned out'?
